load/tracker.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package load
+
+import (
+	"sync"
+	"time"
+)
+
+var _ Tracker[any] = (*DefaultTracker[any])(nil)
+
+type DefaultTracker[T comparable] struct {
+	sync.RWMutex
+
+	txIssuedCounter    uint64
+	txConfirmedCounter uint64
+	txFailedCounter    uint64
+}
+
+func (t *DefaultTracker[T]) Issue(T, time.Time) {
+	t.Lock()
+	t.txIssuedCounter++
+	t.Unlock()
+}
+
+func (t *DefaultTracker[T]) ObserveConfirmed(T) {
+	t.Lock()
+	defer t.Unlock()
+	t.txConfirmedCounter++
+}
+
+func (t *DefaultTracker[T]) ObserveFailed(T) {
+	t.Lock()
+	defer t.Unlock()
+	t.txFailedCounter++
+}
```

Comment:
follow up - we can track latency for each observed containers since we have the time it was issued and when we observe it marked as confirmed/failed